File: .pub-cache/hosted/pub.dartlang.org/camera_android-0.9.8+3/android/src/main/java/io/flutter/plugins/camera/features/flash/FlashMode.java
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package io.flutter.plugins.camera.features.flash;

// Mirrors flash_mode.dart
public enum FlashMode {
  off("off"),
  auto("auto"),
  always("always"),
  torch("torch");

  private final String strValue;

  FlashMode(String strValue) {
    this.strValue = strValue;
  }

  /**
   * Tries to convert the supplied string into a {@see FlashMode} enum value.
   *
   * <p>When the supplied string doesn't match a valid {@see FlashMode} enum value, null is
   * returned.
   *
   * @param modeStr String value to convert into an {@see FlashMode} enum value.
   * @return Matching {@see FlashMode} enum value, or null if no match is found.
   */
  public static FlashMode getValueForString(String modeStr) {
    for (FlashMode value : values()) {
      if (value.strValue.equals(modeStr)) return value;
    }
    return null;
  }

  @Override
  public String toString() {
    return strValue;
  }
}

File: .pub-cache/hosted/pub.dartlang.org/camera_platform_interface-2.2.0/lib/src/types/flash_mode.dart
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

/// The possible flash modes that can be set for a camera
enum FlashMode {
  /// Do not use the flash when taking a picture.
  off,

  /// Let the device decide whether to flash the camera when taking a picture.
  auto,

  /// Always use the flash when taking a picture.
  always,

  /// Turns on the flash light and keeps it on until switched off.
  torch,
}
